src/Lecture3.hs

```diff
@@ -207,11 +261,34 @@ types that can have such an instance.
   implement instances! No spoilers :)
 -}
 
+-- foldable requires a type to be of kind * -> *
+-- only list1 and treasure are of that kind. the rest have arity 0
+-- I thought that List1 not being a monoid would prevent it from being
+-- foldable, but that isn't actually the case. foldMap creates a monoid
+-- out of the structure using the function passed in (which I guess is obvious)
+
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f b (List1 a []) = f a b
+  foldr f b (List1 a (c:as)) = f a $ foldr f b (List1 c as)
```

Comment:
You don't need to implement a manual recursion here, you can reuse the `Foldable` instance for lists 📂